src/Logary/TemplateFormat.fs

```diff
@@ -0,0 +1,101 @@
+﻿[<AutoOpen>]
+module TemplateFormat
+
+open System
+open Logary
+open Utils.FsMessageTemplates
+
+module internal LogaryCapturing =
+  open Logary.Utils.FsMessageTemplates
+
+  let rec convertTemplatePropertyToField (tpv : TemplatePropertyValue) : Value =
+    match tpv with
+    | ScalarValue v ->
+      // TODO: consider types like Guid, DateTime, DateTimeOffset. Are they prematurely stringified in Value.ofObject?
+      // Does that prevent us from using the message template format string later in the pipeline?
+      Value.ofObject v
+
+    | DictionaryValue kvpList ->
+      let stringObjMap =
+        kvpList
+        |> List.map (fun (k, v) ->
+          let keyValue = match k with | ScalarValue v -> v.ToString() | _ -> failwith "only scalar value keys are supported"
+          keyValue, (convertTemplatePropertyToField v))
+        |> Map.ofList
+      Value.Object stringObjMap
+
+    | SequenceValue tpvList ->
+      Value.Array (tpvList |> List.map (convertTemplatePropertyToField))
+
+    | StructureValue (typeTag, pnvList) ->
+      Value.Object (
+        pnvList
+        |> List.map (fun pnv -> pnv.Name, convertTemplatePropertyToField pnv.Value)
+        |> List.append [ "_typeTag", Value.String typeTag ]
+        |> Map.ofList)
+
+  let rec convertToNameAndField (pnv : PropertyNameAndValue) : string * Field =
+    pnv.Name, Field ((convertTemplatePropertyToField pnv.Value), None)
+
+  let capture (template : Utils.FsMessageTemplates.Template) ([<ParamArray>] args : obj[]) =
+    Capturing.captureProperties template args
+    |> Seq.map (convertToNameAndField)
+    |> List.ofSeq
+
+  let logyarDefaultDestructurer : Destructurer =
+    Capturing.createCustomDestructurer (None) (Some Capturing.destructureFSharpTypes)
+
+  let captureField (property : Utils.FsMessageTemplates.Property) value =
+    logyarDefaultDestructurer (DestructureRequest (logyarDefaultDestructurer, value, 10, 0, property.Destr))
+    |> convertTemplatePropertyToField
+    |> fun f -> Field (f, None)
+
+type Message with
```

Comment:
@mavnn Compared to *Xenogloss*, what do you think of this? I think it's a good trade-off in that it doesn't add any dependencies; but there may be easy gain to be had by adding some of your code, perhaps?